src/status_im2/contexts/shell/bottom_tabs.cljs

```diff
@@ -17,35 +18,44 @@
-          :on-press            #(animation/bottom-tab-on-press stack-id)
-          :accessibility-label (str (name stack-id) "-tab"))])
+  ([icon stack-id shared-values notifications-data]
+   [bottom-tab icon stack-id shared-values notifications-data nil])
+  ([icon stack-id shared-values notifications-data double-tap-gesture]
+   [bottom-nav-tab/bottom-nav-tab
+    (cond-> (assoc (get notifications-data stack-id)
+                   :test-ID             stack-id
+                   :icon                icon
+                   :icon-color-anim     (get
+                                         shared-values
+                                         (get shell.constants/tabs-icon-color-keywords stack-id))
+                   :on-press            #(animation/bottom-tab-on-press stack-id)
+                   :accessibility-label (str (name stack-id) "-tab"))
+      double-tap-gesture (assoc :double-tap-gesture double-tap-gesture))]))
 
 (defn bottom-tabs
   []
   [:f>
    (fn []
-     (let [notifications-data (rf/sub [:shell/bottom-tabs-notifications-data])
-           pass-through?      (rf/sub [:shell/shell-pass-through?])
-           shared-values      @animation/shared-values-atom
-           original-style     (style/bottom-tabs-container pass-through?)
-           animated-style     (reanimated/apply-animations-to-style
-                               {:height (:bottom-tabs-height shared-values)}
-                               original-style)]
+     (let [notifications-data          (rf/sub [:shell/bottom-tabs-notifications-data])
+           pass-through?               (rf/sub [:shell/shell-pass-through?])
+           shared-values               @animation/shared-values-atom
+           original-style              (style/bottom-tabs-container pass-through?)
+           animated-style              (reanimated/apply-animations-to-style
+                                        {:height (:bottom-tabs-height shared-values)}
+                                        original-style)
+           messages-double-tap-gesture (-> (gesture/gesture-tap)
+                                           (gesture/number-of-taps 2)
+                                           (gesture/on-start
+                                            (fn [_event]
+                                              (rf/dispatch [:messages-home/select-tab :tab/recent]))))]
        (animation/load-stack @animation/selected-stack-id)
        (reanimated/set-shared-value (:pass-through? shared-values) pass-through?)
        [reanimated/view {:style animated-style}
         (when pass-through?
           [blur/view (blur-overlay-params style/bottom-tabs-blur-overlay)])
         [rn/view {:style (style/bottom-tabs)}
          [bottom-tab :i/communities :communities-stack shared-values notifications-data]
-         [bottom-tab :i/messages :chats-stack shared-values notifications-data]
+         [bottom-tab :i/messages :chats-stack shared-values notifications-data
```

Comment:
its better and simpler to do not implement this into the component, but just wrap bottom-tab here into gesture

Reply:
Done